### ISSUE TITLE
WHERE clauses are silently dropped if there is no FROM

### ISSUE THREAD

**Issue by kosmikus** [haskell-beam/beam#667]
As the title says, a query such as `SELECT true WHERE false` is not rendered correctly by Beam. In general, `WHERE` clauses are dropped in queries where there is no `FROM`. The `WHERE` clause is quite meaningful though, as it determines whether the query has zero or one resulting rows.

This is a minimal example (the real context in which this appeared was significantly more complex):
```haskell
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE KindSignatures #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE OverloadedStrings #-}
{-# OPTIONS_GHC -Wall #-}
module BeamDebug where

import Data.Kind
import Database.Beam
import Database.Beam.Postgres
import Database.Beam.Schema.Tables

data EmptyDb (f :: Type -> Type) =
  EmptyDb

instance Database be EmptyDb where
  zipTables _ _ _ _ = pure EmptyDb

simple :: Q Postgres EmptyDb s (QExpr Postgres s Bool)
simple = do
  guard_ (val_ False)
  pure (val_ True)

test :: IO ()
test = do
  conn <- connectPostgreSQL "dbname=beamtestdb"
  pgTraceStmtIO conn (select simple)

-- output:
-- SELECT true AS "res0"
--
-- expected:
-- SELECT true AS "res0" WHERE false
```

I've been looking at the source code, and the problem seems to be in `Database.Beam.Query.SQL92`:
```haskell
buildSql92Query' :: forall be db s a
                  . ( BeamSqlBackend be, Projectible be a)
                 => Bool {-^ Whether this backend supports arbitrary nested UNION, INTERSECT, EXCEPT -}
                 -> T.Text {-^ Table prefix -}
                 -> Q be db s a
                 -> BeamSqlBackendSelectSyntax be
buildSql92Query' arbitrarilyNestedCombinations baseTblPfx (Q q) =
    buildSelect baseTblPfx (buildQuery baseTblPfx (fromF q))
  where
    be :: Proxy be
    be = Proxy

    buildQuery :: forall s x
                . Projectible be x
               => T.Text
               -> Free (QF be db s) x
               -> SelectBuilder be db x
    buildQuery _ (Pure x) = SelectBuilderQ x emptyQb
    buildQuery tblPfx (Free (QGuard _ next)) = buildQuery tblPfx next
    ...
```
For some reason, "outer" guards are simply dropped, even though they can very much influence the result. I've traced the history of this to 49a6c22679d9bd65f8bf78735fd8ec1ae3b753c3 from March 2017. Before that, it seems this case wasn't present, but the overall structure was significantly different at the time.

I believe the `buildQuery` line should instead be
```haskell
    buildQuery tblPfx f@(Free (QGuard _ _)) = buildJoinedQuery tblPfx f emptyQb
```
but I cannot perfectly judge whether this might introduce any other problems.

I can submit a PR if desired.

**Comment by kmicklas** [comment]
PRs welcome! We should definitely fix this.

**Comment by LaurentRDC** [comment]
Fixed via #695
